api.yaml

```diff
@@ -2219,6 +2277,20 @@ components:
           format: binary
       x-swagger-mongoose:
         exclude-schema: true
+    set_role_body:
+      required:
+      - role
+      type: object
+      properties:
+        role:
+          type: string
+          example: teacher
+          description: |
+            The role for the user to have. Some
+            examples include teacher, student 
+            teaching-assistant or researcher
+      x-swagger-mongoose:
+        exclude-schema: true
```

Comment:
Usually this is just a simple property not an object. In addition, OpenAPI has the concept of enum, so it is possible to define a restricted set of values (e.g. teacher, student, etc.)

Reply:
fixed